Library/Homebrew/upgrade.rb

```diff
@@ -274,8 +285,7 @@ def self.check_installed_dependents(
         end
         return
       end
-
-      installed_formulae = (dry_run ? formulae : FormulaInstaller.installed.to_a).dup
+      installed_formulae = formulae.dup
       installed_formulae.reject! { |f| f.core_formula? && f.versioned_formula? }
       return if installed_formulae.empty?
```

Comment:
This sounds like it'll now upgrade dependents of things that failed to upgrade.

We may need to group dependents by formula and then filter out those that didn't actually install later.